build.assets/version-check/main.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// Command version-check-latest exits non-zero when given a
+// git tag that is not greater than the highest semver ever
+// published on GitHub Releases. This allows us to avoid updating
+// "latest" packages or tags when publishing patch releases for
+// older branches.
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"sort"
+	"strings"
+	"time"
+
+	"golang.org/x/mod/semver"
+
+	"github.com/gravitational/trace"
+
+	go_github "github.com/google/go-github/v41/github"
+)
+
+func main() {
+	tag, check, err := parseFlags()
+	if err != nil {
+		log.Fatalf("Failed to parse flags: %v.", err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	switch check {
+	case "latest":
+		err = checkLatest(ctx, tag, newGitHub())
+	case "prerelease":
+		err = checkPrerelease(tag)
+	default:
+		log.Fatalf("invalid check: %v", check)
+	}
+
+	if err != nil {
+		log.Fatalf("Check failed: %v.", err)
+	}
+}
+
+func parseFlags() (string, string, error) {
+	tag := flag.String("tag", "", "tag to validate")
+	check := flag.String("check", "", "check to run [latest, prerelease]")
+	flag.Parse()
+
+	if *tag == "" {
+		return "", "", trace.BadParameter("tag missing")
+	}
+	if *check == "" {
+		return "", "", trace.BadParameter("check missing")
+	}
+	switch *check {
+	case "latest":
+		break
+	case "prerelease":
+		break
+	default:
+		return "", "", trace.BadParameter("invalid check: %v", *check)
+	}
```

Comment:
```suggestion
	switch *check {
	case "latest", "prerelease":
	default:
		return "", "", trace.BadParameter("invalid check: %v", *check)
	}
```

You can combine the two cases, and you don't need a break because go switches don't fallthrough by default.

Reply:
Fixed in de0f1d4.